### PR TITLE
Wallet, Account, Balances and Asset Transfer implementation

### LINKED ISSUE #26: wallet: create, import, recover, and export
As a developer, I would like to have methods on the `wallet` module (convenience layer, discussed by #25) that I could:
* create new addresses to a wallet (not only NEP-6 and NEP-2, but also with BIP-39)
* import a wallet/address (NEP-6, NEP-2, .json format)
* recover a wallet/address (NEP-6, NEP-2, BIP-39, .json format)
* export (backup) a wallet/address (NEP-6, NEP-2, .json format)

Most of these functionalities are kind of wrappers (abstraction layer) to the methods implemented on the `core` and/or `crypto` modules.

### PR DESCRIPTION
**Pull request by claudemiller**
Resolves #26, #27, #29.

### CODE REVIEW

File: wallet/src/main/java/io/neow3j/wallet/AssetTransfer.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+            }
+            byte[] rawUnsignedTx = tx.toArray();
+            ECKeyPair keyPair = account.getECKeyPair();
+
+            RawInvocationScript invocationScript = new RawInvocationScript(
+                    Sign.signMessage(rawUnsignedTx, keyPair));
+
+            String publicKeyHexString = Numeric.toHexStringNoPrefix(account.getPublicKey());
+            byte[] publicKeyByteArray = Numeric.hexStringToByteArray(publicKeyHexString);
+            RawVerificationScript verificationScript = Keys.getVerificationScriptFromPublicKey(publicKeyByteArray);
```
Comment (gsmachado) [inline wallet/src/main/java/io/neow3j/wallet/AssetTransfer.java]: Simplify?! :-)

File: wallet/src/main/java/io/neow3j/wallet/Balances.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    /**
+     * Overrides the asset balances with the provided information in the unspents.
+     * @param unspents The unspents fetched from a RPC node.
+     */
+    public void updateAssetBalances(NeoGetUnspents.Unspents unspents) {
+        if (!unspents.getAddress().equals(account.getAddress())) {
+            throw new IllegalArgumentException("The provided unspents are not related to the " +
+                    "account of this balance");
+        }
+        assetBalances = new HashMap<>();
```
Comment (gsmachado) [inline wallet/src/main/java/io/neow3j/wallet/Balances.java]: Can we infer the assetBalances size? ;-)

File: wallet/src/main/java/io/neow3j/wallet/Balances.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    /**
+     * Overrides the token balances with the provided balances.
+     * @param balances The NEP5 token balances fetched from a RPC node.
+     */
+    public void updateTokenBalances(NeoGetNep5Balances.Balances balances) {
+        if (!balances.getAddress().equals(account.getAddress())) {
+            throw new IllegalArgumentException("The provided token balances are not related to the " +
+                    "account of this balance");
+        }
+        this.tokenBalances = new HashMap<>();
```
Comment (gsmachado) [inline wallet/src/main/java/io/neow3j/wallet/Balances.java]: Infer the size?! Same as above.

File: wallet/src/test/java/io/neow3j/wallet/WalletTest.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    @Test
+    public void testCreateWalletFromNEP6File() throws IOException {
+
+        URL nep6WalletFile = Thread.currentThread().getContextClassLoader().getResource("wallet.json");
+        ObjectMapper mapper = new ObjectMapper();
+        NEP6Wallet nep6Wallet = mapper.readValue(nep6WalletFile, NEP6Wallet.class);
+
+        Wallet w = Wallet.fromNEP6Wallet(nep6Wallet).build();
```
Comment (gsmachado) [inline wallet/src/test/java/io/neow3j/wallet/WalletTest.java]: As discussed, create a method called "load" or something like this, that will get as parameter a String or Path with the path to the wallet (.json).

File: wallet/src/main/java/io/neow3j/wallet/Account.java [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+            this.encryptedPrivateKey= NEP2.encrypt(password, getECKeyPair(), scryptParams);
+        }
+    }
+
+    public boolean isMultiSig() {
+        if (contract != null && contract.getScript() != null) {
+            String script = contract.getScript();
+            return script.substring(script.length() - 2).equals(OpCode.toHexString(CHECKMULTISIG));
+        }
+        return false;
```
[the remark this answers is not in the document]
Comment (claudemiller) [reply wallet/src/main/java/io/neow3j/wallet/Account.java]: I can currently not determine if it is a multi-sig if the verification script is not given. We could allow an account to hold more than one public key, easily identifying it as a multi-sig account. 
But I wish to first get feedback from developers or the NGF about usage of multi-sig accounts. Maybe we don't need to spend so much though on it.

File: wallet/src/test/java/io/neow3j/wallet/WalletTest.java (hunk 4 above)
Comment (claudemiller) [reply wallet/src/test/java/io/neow3j/wallet/WalletTest.java]: I added a method that takes an URL object. I guess usually one gets a resource via `SomeClass.class.getResource("some_resource")` which returns an URL.

**Comment by gsmachado** [comment]
It's fine now. :-) Great work!

**Comment by gsmachado** [comment]
It's fine now. :-) Great work!

**Pull request merged**